### PR TITLE
[9.0][ADD] gdpr_partner_report

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
This PR adds:

- Improvements on `.gitignore` and `.travis.yml` files.
- `data_protection_base` module and `gdpr_base` modules (Enable menus).
- `gdpr_partner_report` module.

GDPR Partner Report
===================

This module helps a company to identify all the transactions that a specific
partner is involved in, with the possibility to export the associated data.

- [ ] Depends on https://github.com/OCA/OCB/pull/753.

### CODE REVIEW

**Comment by yajo** [comment]
Some questions about module naming:

1. Should we prefix addons with some shorter convention, such as `dp_`, or `privacy_`?
1. I feel that we should avoid the `gdpr_` prefix. Just to put an example, in Spain we had LOPD, then RGPD, then now GDPR. Naming the addon after each acronym would have been a nightmare, and narrowing its reach to just one jurisdiction also seems innecesary (think of those outside UE that want to use it). Isn't it better to stick with the prefix we choose at point 1?

**Comment by MiquelRForgeFlow** [comment]
Yeah. No problem with renaming, but I want maximum consensus before doing it. I don't want to change the name four times  😏

**Comment by pedrobaeza** [comment]
My vote for `privacy_`. We don't want more initials to get confused (dp: decimal precision? :laughing: )

**Comment by yajo** [comment]
It seems it's most accepted in the [google doc](https://docs.google.com/document/d/1UsaO0hpJb21oQ6l-2YZ9LroJ3k3zxLtAmbVhsb6p1X8); let's use `privacy_` then.

I'll do further review after rename, or GH will trash all comments

**Comment by MiquelRForgeFlow** [comment]
I don't know what fails in Travis, but Runbot is :green_apple: 😎

As you see in the Readme, the xlsx report doesn't work because the pending fix in the odoo.

File: data_protection_base/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    'summary': 'Provides security and menus for data protection modules.',
+    'author': "Eficent, "
+              "Odoo Community Association (OCA)",
+    'website': 'http://www.github.com/Eficent/gdpr,',
+    'license': 'AGPL-3',
+    'data': [
+        'security/data_protection.xml',
+        'views/data_protection_menu_view.xml',
+    ],
+    'installable': True,
```
Comment (dreispt) [inline data_protection_base/__openerp__.py:17]: Add "application: True". This should the the App where further features will be anchored.

File: data_protection_base/views/data_protection_menu_view.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        <field name="res_model">res.partner</field>
+        <field name="view_type">form</field>
+        <field name="view_mode">tree,form</field>
+        <field name="domain">[]</field>
+        <field name="filter" eval="True"/>
+    </record>
+
+    <menuitem id="parent_menu_data_protection"
+              name="Data Protection"
+              groups="group_data_protection_user,group_data_protection_manager"
```
Comment (dreispt) [inline data_protection_base/views/data_protection_menu_view.xml:19]: User should be enough, since Manager inherits User permissions

**Review by dreispt — COMMENTED** [review]
Both "gdpr_base" and "gdpr_protection_base" should be a single module named "privacy_base". A "privacy_partner_report" should be proposed in a separate PR.

**Comment by MiquelRForgeFlow** [comment]
When https://github.com/OCA/data-protection/pull/2 gets merged, then I will open a new PR for the partner report.

**Comment by dreispt** [comment]
@mreficent no point in waiting, please go ahead asap. You can fix the dependency name for the base module once #1 is sorted out.

**Pull request closed without merging**